src/modules/notification/infrastructure/store/notification.store.ts

```diff
@@ -1,23 +1,9 @@
-import EventNotification from '../../domain/entity/EventNotification';
-import valtioEventNotificationStore, { EventNotificationState, subscribe } from './valtio/notification.store.valtio';
+import { createStore } from 'vuex';
+import storeDefinition from './modules/notification';
+import { Store } from './types';
 
-export type StoreListener = (state: EventNotificationState) => void;
+export const store = createStore(storeDefinition);
 
-class EventNotificationStore {
```

Comment:
This particular store is supposed to be the Facade sitting between the store implementation details (Vuex for example) and the Application Layer (the hooks). The idea of this Facade is the Application Layer does not care about the implementation details of the store. All it has to do is to ask the Facade specific things.

You can understand it better via a simple real-world analogy:

![](https://refactoring.guru/images/patterns/diagrams/facade/live-example-en.png?id=461900f9fbacdd0ce981)

When you call a shop to place a phone order, an operator is your facade to all services and departments of the shop. The operator provides you with a simple voice interface to the ordering system, payment gateways, and various delivery services.

What you are doing here (by directly exporting Vuex's store) is breaking the Facade. Now the customer needs to know about the details of all the delivery services (meaning now the hook needs to know about all the methods of a Vuex's store):

![image](https://user-images.githubusercontent.com/24670327/115810809-c23ed480-a418-11eb-83c6-a88b070e9c8f.png)

Or in other words, you're still coupling your Application Layer to the store's implementation details.